packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -0,0 +1,185 @@
+///
+/// Example: [12 | 03]
+class CupertinoCountdownTimerPicker extends StatefulWidget {
+  /// Constructs an iOS style countdown timer picker.
+  ///
+  /// [onTimerDurationChanged] is the callback when the selected duration changes
+  /// and must not be null.
+  ///
+  /// [initialTimerDuration] defaults to 1 minute and is limited from 1 minute
+  /// to 23 hours 59 minutes. Only hour and minute values are extracted
+  /// from [initialTimerDuration], so specifying other fields like day, second,
+  /// etc. will not affect anything.
+  ///
+  /// [minuteInterval] is the granularity of the minute spinner. Must be a factor
+  /// of 60.
+  CupertinoCountdownTimerPicker({
+    this.initialTimerDuration = const Duration(minutes: 1),
+    this.minuteInterval = 1,
+    @required this.onTimerDurationChanged,
+  }) : assert(onTimerDurationChanged != null),
+       assert(initialTimerDuration >= const Duration(minutes: 1)),
+       assert(initialTimerDuration < const Duration(days: 1)),
+       assert(60 % minuteInterval == 0);
+
+  /// The initial duration of the countdown timer.
+  final Duration initialTimerDuration;
+
+  /// The granularity of the minute spinner. Must be a factor of 60.
+  final int minuteInterval;
+
+  /// Callback when the timer duration changes.
+  final ValueChanged<Duration> onTimerDurationChanged;
+
+  @override
+  State<StatefulWidget> createState() => _CountdownTimerState();
+}
+
+class _CountdownTimerState extends State<CupertinoCountdownTimerPicker> {
+  int _selectedHour;
+  int _selectedMinute;
+
+  FixedExtentScrollController hourController;
+  FixedExtentScrollController minuteController;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _selectedHour = widget.initialTimerDuration.inHours;
+    _selectedMinute = widget.initialTimerDuration.inMinutes % 60;
+
+    hourController = new FixedExtentScrollController(initialItem: _selectedHour);
+    minuteController = new FixedExtentScrollController(
+      initialItem: _selectedMinute ~/ widget.minuteInterval);
+  }
+
+  Widget _buildHourPicker(BuildContext context) {
+    return new CupertinoPicker(
+      scrollController: hourController,
+      offAxisFraction: -0.15,
+      useMagnifier: _kUseMagnifier,
+      magnification: _kMagnification,
+      itemExtent: _kItemExtent,
+      backgroundColor: _kBackgroundColor,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          _selectedHour = index;
+          widget.onTimerDurationChanged(
+            new Duration(hours: _selectedHour, minutes: _selectedMinute));
+        });
+      },
+      children: new List<Widget>.generate(24, (int index) {
+        return new Container(
+          alignment: Alignment.center,
+          child: new Text(index.toString().padLeft(2, '  ')),
+        );
+      }),
+    );
+  }
+
+  Widget _buildMinutePicker(BuildContext context) {
+    return new CupertinoPicker(
+      scrollController: minuteController,
+      offAxisFraction: 0.5,
```

Comment:
Note: when you try out different text scales, it's possible that will affect how this should look. If that's the case, you may need to write a method to calculate what these values should be.